test/e2e/functional_test.go

```diff
@@ -84,6 +84,31 @@ spec:
 		ExpectWorkflowDeleted()
 }
 
+func (s *FunctionalSuite) TestWorkflowRetention() {
+	var listOptions metav1.ListOptions
+	for i := 0; i < 15; i++ {
```

Comment:
Spawning 15 worklfows sounds absurdly expensive... For an E2E test like this, you should set the maximum to 1, then spawn 2 workflows

Reply:
How do I set the default for the test infra to consider since we don't want to have any defaults?

Reply:
Set them in the quick start, they should be good defaults for the real world, as users will take them as examples. For e2e tests, you can add a kustomise patch, see test/e2e/manifests

Reply:
I will update this after testing some stuff.